forc-util/src/restricted.rs

```diff
@@ -62,7 +62,7 @@ pub fn contains_invalid_char(name: &str, use_case: &str) -> Result<()> {
         }
     }
     for ch in chars {
-        if !(unicode_xid::UnicodeXID::is_xid_continue(ch) || ch == '-') {
+        if !(unicode_xid::UnicodeXID::is_xid_continue(ch) || ch == '-' || ch == '/') {
```

Comment:
This function checks for valid package names, and package names shouldn't contain `'/'` so I don't think we want this.

Reply:
@mitchmindtree Is that not what you mentioned [here](https://github.com/FuelLabs/sway/pull/1669#pullrequestreview-987011928) and [here](https://github.com/FuelLabs/sway/pull/1669#discussion_r883256924)?

Reply:
In [this comment](https://github.com/FuelLabs/sway/pull/1669#discussion_r883256924) my intention was to suggest that the new command should support full paths (i.e. `forc new foo/bar/proj-name`) rather than only supporting a project name (i.e. `forc new proj-name`).

That suggestion is only related to this function in the sense that the default project name used for `forc new` is retrieved from the directory name. In the command `forc new foo/bar/baz`, the inferred project name is not `foo/bar/baz`, but rather `baz` (taken from the directory name, not the full path). As a result, there should be no need for this project name validation function to support `/` characters.

Hopefully this makes a bit more sense?

Reply:
@mitchmindtree Ah, I see. Change incoming